src/app/store/products-store/products.reducer.ts

```diff
@@ -0,0 +1,40 @@
+import { createReducer, on } from '@ngrx/store';
+import ProductInterface from '../../interfaces/products';
+import {
+  searchProducts,
+  searchProductsFailure,
+  searchProductsSuccess,
+} from './products.actions';
+
+export const productsFeatureKey = 'Products';
+
+export interface ProductState {
+  products: ProductInterface[];
+  error: string | null;
+  status: 'idle' | 'loading' | 'success' | 'failure';
+}
+
+export const initialState: ProductState = {
+  products: [],
+  error: null,
+  status: 'idle',
+};
+
+export const productsReducer = createReducer(
+  initialState,
+  on(searchProducts, (state) => ({
+    ...state,
```

Comment:
Maybe make `error: null` here too 🤔 Not sure if it makes sense though (as in, is it necessary)